src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -249,8 +249,15 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
             var filePath = file.Path;
             try
             {
-                using (var data = OpenFileForReadWithSmallBufferOptimization(filePath))
+                if (file.IsInputRedirected)
                 {
+                    using var data = Console.OpenStandardInput(bufferSize: 1);
```

Comment:
>bufferSize: 1 [](start = 63, length = 13)

Just curious: Why set an explicit `bufferSize`?

Reply:
I used the same buffer size as in case of FileStream per this comment:
https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L342-L347

Reply:
Does that optimization apply to streams where `Seek()` is not supported?

---
In reply to: [373165431](https://github.com/dotnet/roslyn/pull/41166#discussion_r373165431) [](ancestors = 373165431)

Reply:
Good point, and https://github.com/dotnet/runtime/blob/b3fa13f/src/libraries/System.Console/src/System/Console.cs#L371-L379 says it all. So I will remove the bufferSize. 😅